### PR TITLE
Volodin Evgeniy. Lab №1. Option №1.

### PR DESCRIPTION
**Pull request by evldn**
In this lab, a plugin was implemented that prints the names of all classes (structures) and their fields.
Example:
```C++ 
class MyClass {
    int a;
    float b;
};
```
Result:
```
MyClass
 |_a
 |_b`
```

### CODE REVIEW

File: clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  explicit PrintClassVisitor(ASTContext *Context) : Context(Context) {}
+
+  bool VisitCXXRecordDecl(CXXRecordDecl *declaration) {
+    llvm::outs() << declaration->getNameAsString().c_str() << "\n";
+    for (const auto &field : declaration->fields()) {
+      llvm::outs() << " |_" << field->getNameAsString().c_str() << "\n";
+    }
+    return true;
+  }
+  ASTContext *Context;
```
Comment (m-ly4) [inline clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp]: Is it necessary here?

File: clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  PrintClassVisitor Visitor;
+};
+
+class PrintClassASTAction : public PluginASTAction {
+public:
+  std::unique_ptr<clang::ASTConsumer>
+  CreateASTConsumer(clang::CompilerInstance &Compiler,
+                    llvm::StringRef InFile) override {
+    return std::unique_ptr<clang::ASTConsumer>(
+        new PrintClassConsumer(Compiler));
```
Comment (m-ly4) [inline clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp]: Could you please make a smart ptr without explicit `new` call?

File: clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  bool ParseArgs(const CompilerInstance &CI,
+                 const std::vector<std::string> &args) override {
+    return true;
```
Comment (m-ly4) [inline clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp]: Could you please add and test `help` option for the plugin?

**Review by m-ly4 — CHANGES_REQUESTED** [review]

**Comment by evldn** [comment]
The following changes have been made:
1. The ASTContext *Context field was removed from the PrintClassVisitor class as unnecessary.
2. The CreateASTConsumer method of the PrintClassLASTAction class now returns make_unique, not new.
3. The help option for the plugin has been added and tested.

**Review by m-ly4 — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

**Pull request merged**